src/OSPSuite.Core/Domain/Services/SpatialStructureMerger.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Linq;
+using OSPSuite.Assets;
+using OSPSuite.Core.Domain.Builder;
+using OSPSuite.Core.Domain.Descriptors;
+using OSPSuite.Core.Domain.Mappers;
+using OSPSuite.Utility.Exceptions;
+using OSPSuite.Utility.Extensions;
+
+namespace OSPSuite.Core.Domain.Services
+{
+   internal interface ISpatialStructureMerger
+   {
+      IContainer MergeContainerStructure(ModelConfiguration modelConfiguration);
+      IContainer MergeNeighborhoods(ModelConfiguration modelConfiguration);
+   }
+
+   internal class SpatialStructureMerger : ISpatialStructureMerger
+   {
+      private readonly IObjectBaseFactory _objectBaseFactory;
+      private readonly IContainerBuilderToContainerMapper _containerMapper;
+      private readonly INeighborhoodCollectionToContainerMapper _neighborhoodsMapper;
+
+      public SpatialStructureMerger(
+         IObjectBaseFactory objectBaseFactory,
+         IContainerBuilderToContainerMapper containerMapper,
+         INeighborhoodCollectionToContainerMapper neighborhoodsMapper)
+      {
+         _objectBaseFactory = objectBaseFactory;
+         _containerMapper = containerMapper;
+         _neighborhoodsMapper = neighborhoodsMapper;
+      }
+
+      public IContainer MergeContainerStructure(ModelConfiguration modelConfiguration)
+      {
+         var (model, simulationBuilder) = modelConfiguration;
+         // Create Root Container for the model with the name of the model
+         var root = _objectBaseFactory.Create<IContainer>()
+            .WithName(model.Name)
+            .WithMode(ContainerMode.Logical)
+            .WithContainerType(ContainerType.Simulation);
+
+         root.AddTag(new Tag(Constants.ROOT_CONTAINER_TAG));
+
+         createMergedContainerStructureInRoot(root, simulationBuilder);
+
+         return root;
+      }
+
+      private void createMergedContainerStructureInRoot(IContainer root, SimulationBuilder simulationBuilder)
```

Comment:
@rwmcintosh @Yuri05 Please review this. only a few lines of code dealing with merging, replacing etc...I think this is what we want